modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,151 +557,197 @@ sub print_diagram {
             }
 
+
         } elsif($tweak=~/^analysis\[([^\]]+)\]\.(wait_for|flow_into)(\?|#|\+?=(.+))$/) {
+
             my ($analyses_pattern, $attrib_name, $operation, $new_value_str) = ($1, $2, $3, $4);
             $operation=~/^(\?|#|\+?=)/;
             my $operator = $1;
 
             my $analyses = $self->collection_of( 'Analysis' )->find_all_by_pattern( $analyses_pattern );
-            print "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
+            push @response, "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
 
             my $new_value = destringify( $new_value_str );
 
             foreach my $analysis (@$analyses) {
-
+                my $tweakStructure;
+                $tweakStructure->{Object}->{Type} = "Analysis";
                 my $analysis_name = $analysis->logic_name;
-
+                $tweakStructure->{Object}->{Id} = $analysis->dbID + 0;
+                $tweakStructure->{Object}->{Name} = $analysis->logic_name;
                 if( $attrib_name eq 'wait_for' ) {
-
+                    $tweakStructure->{Return}->{Field} = 'wait_for';
                     my $cr_collection   = $self->collection_of( 'AnalysisCtrlRule' );
                     my $acr_collection  = $analysis->control_rules_collection;
-
+                    $tweakStructure->{Return}->{OldValue} = join(', ', map { $_->condition_analysis_url } @$acr_collection );
                     if($operator eq '?') {
-                        print "Tweak.Show    \tanalysis[$analysis_name].wait_for ::\t[".join(', ', map { $_->condition_analysis_url } @$acr_collection )."]\n";
+                        $tweakStructure->{Action} = "SHOW";
+                        $tweakStructure->{Return}->{NewValue} = $tweakStructure->{Return}->{OldValue};
+                        push @response, "Tweak.Show  \tanalysis[$analysis_name].wait_for ::\t[".join(', ', map { $_->condition_analysis_url } @$acr_collection )."]\n";
                     }
 
                     if($operator eq '#' or $operator eq '=') {     # delete the existing rules
+                        $tweakStructure->{Action} = "DELETE";
+                        $tweakStructure->{Return}->{NewValue} = undef;
                         foreach my $c_rule ( @$acr_collection ) {
                             $cr_collection->forget_and_mark_for_deletion( $c_rule );
                             $need_write = 1;
 
-                            print "Tweak.Deleting\t".$c_rule->toString." --> (missing value)\n";
+                            push @response, "Tweak.Deleting\t".$c_rule->toString." --> (missing value)\n";
                         }
                     }
 
                     if($operator eq '=' or $operator eq '+=') {     # create new rules
+                        $tweakStructure->{Action} = "SET";
+                        $tweakStructure->{Return}->{NewValue} = $new_value;
```

Comment:
When the action is `+=`, _Action_ will be `SET` and _NewValue_ the stuff that is **added**. That doesn't feel right. _NewValue_ should be the new, complete, value of the field, so should be set after calling `parse_wait_for` in the same way as `OldValue`. I wonder whether _Action_ should be something like `ADD` or `EXTEND` (must be consistent across all tweak syntaxes)

Reply:
The same with actions above - they have unusual structure